go/internal/feast/featurestore.go

```diff
@@ -20,17 +26,24 @@ type FeatureStore struct {
 	onlineStore OnlineStore
 }
 
-type entityKeyRow struct {
-	entityKey  *types.EntityKey
-	rowIndices []int
-}
-
-// A Features struct specifies a list of features to be retrieved from the online store. These features
+// A Features struct specifies a list of Features to be retrieved from the online store. These Features
 // can be specified either as a list of string feature references or as a feature service. String
 // feature references must have format "feature_view:feature", e.g. "customer_fv:daily_transactions".
 type Features struct {
-	features       []string
-	featureService *FeatureService
+	Features       []string
+	FeatureService *FeatureService
+}
+
+type FeatureVector struct {
+	Name       string
+	Values     array.Interface
+	Statuses   []serving.FieldStatus
+	Timestamps []*timestamppb.Timestamp
+}
+
+type featuresAndView struct {
+	view     *FeatureView
+	features []string
 }
 
 type GroupedFeaturesPerEntitySet struct {
```

Comment:
nit: can we have a comment for this struct?

Reply:
added